state/immutable-trie/trie.go

```diff
@@ -143,16 +143,11 @@ func (t *Trie) Commit(objs []*state.Object) (state.Snapshot, []byte) {
 			}
 
 			if len(obj.Storage) != 0 {
-				localSnapshot, err := t.state.NewSnapshotAt(obj.Root)
+				trie, err := t.state.newTrieAt(obj.Root)
 				if err != nil {
 					panic(err)
```

Comment:
could we return the error with some context?

Reply:
Do you mean to remove the panic? or throw a panic with some context? The error already includes the root that cannot be found. I would like to remove the panic in another PR since it was already there. I have tried it but many parts of the code change.

Reply:
as a context I mead to use something like `fmt.Errorf("%w on root %s", err, obj.Root)`